src/components/Input/index.test.tsx

```diff
@@ -0,0 +1,189 @@
+import {
+  render,
+  screen,
+  fireEvent,
+  within,
+  cleanup,
+} from "@testing-library/react";
+import Input from "./index";
+import { Suggestion, SuggestionList } from "./styledComponents";
+
+describe("Input component", () => {
+  it("should allow letters to be inputted", () => {
+    render(
+      <Input
+        suggestions={[
+          "Alligator",
+          "Bask",
+          "Crocodilian",
+          "Death Roll",
+          "Eggs",
+          "Jaws",
+          "Reptile",
+          "Solitary",
+          "Tail",
+          "Wetlands",
+        ]}
+      />
+    );
+
+    let input = screen.getByLabelText("search-input") as HTMLInputElement;
+    expect(input.value).toBe("");
+    fireEvent.change(input, { target: { value: "Good Day" } });
+    expect(input.value).toBe("Good Day");
+  });
+
+  it("should show suggestions list matching input target", () => {
+    render(
+      <Input
+        suggestions={[
+          "Alligator",
+          "Bask",
+          "Crocodilian",
+          "Death Roll",
+          "Eggs",
+          "Jaws",
+          "Reptile",
+          "Solitary",
+          "Tail",
+          "Wetlands",
+        ]}
+      />
+    );
+    let input = screen.getByLabelText("search-input") as HTMLInputElement;
+
+    expect(input.value).toBe("");
+    fireEvent.change(input, { target: { value: "e" } });
+    expect(input.value).toBe("e");
+
+    const list = screen.getByRole("list");
+    const { getAllByRole } = within(list);
+    const items = getAllByRole("listitem");
+    expect(items.length).toBe(4);
+  });
+
+  it("should show in input field chosen suggestion", () => {
+    render(
+      <Input
+        suggestions={[
+          "Alligator",
+          "Bask",
+          "Crocodilian",
+          "Death Roll",
+          "Eggs",
+          "Jaws",
+          "Reptile",
+          "Solitary",
+          "Tail",
+          "Wetlands",
+        ]}
+      />
+    );
+    let input = screen.getByLabelText("search-input") as HTMLInputElement;
+
+    expect(input.value).toBe("");
+    fireEvent.change(input, { target: { value: "All" } });
+    expect(input.value).toBe("All");
+
+    const list = screen.getByRole("list");
+    const { getAllByRole } = within(list);
+    const items = getAllByRole("listitem");
+
+    expect(items.length).toBe(1);
+
+    const item = screen.getByText("Alligator").textContent;
+
+    fireEvent.change(input, { target: { value: item } });
+    expect(input.value).toBe("Alligator");
+  });
+
+  it("shouldn't show remove button in input field", () => {
```

Comment:
These two tests (from line 100 and 125) can be merged into one. Maybe named as "it shows remove button after suggestion list is displayed"?